### ISSUE TITLE
Check certs expiration before restoring snapshot

### ISSUE THREAD

**Issue by maggieliu** [rancher/rke#1928]
Before restoring etcd snapshot via RKE CLI, check if certs are expiring. If can't find cert or it's expired exit restoration.

**Comment by deniseschannon** [comment]
Available with RKE v1.1.0-rc16

**Comment by sowmyav27** [comment]
**Verified with rke `v1.1.0-rc16`**

1. Deploy a cluster expired with certs (using rke cli binary with certs expiry in 15 minutes)
2. Take a snapshot
3. Wait for the cluster certs to expire
4. restore from snapshot (taken in step 2) using rke cli 1.1.0-rc16 gives me error - 
```
./rke etcd snapshot-restore --name test_certs --config rancher-cluster.yml
WARN[0000] This is not an officially supported version (v1.1.0-rc16) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest 
INFO[0000] Running RKE version: v1.1.0-rc16             
INFO[0000] Restoring etcd snapshot test_certs           
FATA[0000] [etcd] Failed to restore etcd snapshot: invalid certs: Certificate [kube-node] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-4-104] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-scheduler] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-0-167] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-controller-manager] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-proxy] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-6-140] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-ca] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver-proxy-client] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver-proxy-client] failed verification: x509: certificate has expired or is not yet valid 
```
5. Rotate certs - `./rke cert rotate --config rancher-cluster.yml` (rotates all certs except kube-ca)
6. Error seen when trying to restore to snapshot (taken in step 2) - 
```
./rke etcd snapshot-restore --name test_certs --config rancher-cluster.yml
WARN[0000] This is not an officially supported version (v1.1.0-rc16) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest 
INFO[0000] Running RKE version: v1.1.0-rc16             
INFO[0000] Restoring etcd snapshot test_certs           
FATA[0000] [etcd] Failed to restore etcd snapshot: invalid certs: Certificate [kube-ca] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-ca] failed verification: x509: certificate has expired or is not yet valid 
```
7. Do a cert rotate with ca -- `./rke cert rotate --rotate-ca --config rancher-cluster.yml`
8. No error seen while doing a snapshot restore now.
9. `kubectl get nodes --kubeconfig kube_config_rancher-cluster.yml ` works. and lists the nodes of cluster
